lib/pages.js

```diff
@@ -0,0 +1,63 @@
+const { generateNode } = require('./nodes');
+const fetch = require('node-fetch').default;
+
+module.exports = {
+  getPageIds: async function (routeBase, opts) {
+    try {
+      const response = await fetch(`${routeBase}/@apostrophecms/page?apikey=${opts.apiKey}&all=1&flat=1`);
+      const data = await response.json();
+      const pages = data.results.filter(doc => doc.type !== '@apostrophecms/trash');
+      return pages.map(page => page._id);
+    } catch (error) {
+      console.error('Error requesting Apostrophe page IDs:', error);
+      return [];
+    }
+  },
+  generatePageNodes: async function (ids, {
+    baseUrl, apiRouteBase, apiKey, renderPages
+  }, utils) {
+    for (const id of ids) {
+      let page;
+
+      try {
+        const response = await fetch(`${apiRouteBase}/@apostrophecms/page/${id}?apikey=${apiKey}`);
+        page = await response.json();
+      } catch (error) {
+        console.error(`Error requesting Apostrophe page with ID ${id}`, error);
+        return;
+      }
+
+      if (!page || page.message === 'notfound') {
+        // This is most likely the trash "page"
+        console.warn(`No page found for Apostrophe document ID ${id}.`);
+        return;
+      }
+
+      if (renderPages !== false && page._url) {
+        await addRenderedContent(page, baseUrl);
```

Comment:
You might want this for a piece that has a _url, too. I agree it is somewhat more probable in the case of a page.

Reply:
I think we should hold off on that for now. For one, I'm thinking of the Gatsby use case. It seems to be all about that GraphQL. Simply delivering rendered HTML doesn't seem to be The Way. It also adds a ton of more requests. I'd rather this be happening on the Apostrophe end if we want to do more rendering.